File: php/testcode/benchmark_test_00336.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00336(BenchmarkRequest $req): BenchmarkResponse {
    $ip = $req->param('ip');
    if (!preg_match('/^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$/', $ip)) {
        return BenchmarkResponse::badRequest('invalid ip');
    }
    $output = [];
    exec("ping -c 1 $ip", $output);
    return BenchmarkResponse::ok(implode("\n", $output));
}

File: rust/testcode/benchmark_test_00336.rs
fn authenticate_user(username: &str) -> super::shared::BenchmarkResponse {
    super::shared::BenchmarkResponse::ok(&format!("logged in as {}", username))
}

pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let username = req.param("username");
    let _password = req.param("password");

    let response = authenticate_user(&username);

    response
}
